cpp/tests/test_sde_sir.cpp

```diff
@@ -0,0 +1,166 @@
+        model.parameters.set<mio::ssir::TransmissionProbabilityOnContact>(3);
+        model.parameters.get<mio::ssir::ContactPatterns>().get_baseline()(0, 0) = 1;
+    }
+    return model;
+}
+
+TEST(TestSdeSir, get_flows)
+{
+    // make two get_flows computations with mocked rng
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(4))
+        .WillOnce(testing::Return(1.))
+        .WillOnce(testing::Return(0.))
+        .WillOnce(testing::Return(0.))
+        .WillOnce(testing::Return(1.));
+
+    const Eigen::Vector3d y = Eigen::Vector3d::Constant(1);
+    Eigen::Vector2d flows   = Eigen::Vector2d::Constant(1);
+
+    // results contain two parts : deterministic + stochastic
+
+    ssir_testing_model().get_flows(y, y, 0, flows);
+    auto expected_result = Eigen::Vector2d{1 + 2, 1};
+    EXPECT_EQ(flows, expected_result);
+
+    ssir_testing_model().get_flows(y, y, 0, flows);
+    expected_result = Eigen::Vector2d{1, 1 + 2};
+    EXPECT_EQ(flows, expected_result);
+}
+
+TEST(TestSdeSir, Simulation)
+{
+    // make a single integration step via a simulation
+    // this should overwrite the model step size
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(2))
+        .WillOnce(testing::Return(.5))
+        .WillOnce(testing::Return(.5));
+
+    auto integrator = std::make_shared<mio::EulerIntegratorCore>();
+    auto sim        = mio::ssir::Simulation(ssir_testing_model(), 0, 1);
+    sim.set_integrator(integrator);
+    sim.advance(1);
+
+    EXPECT_EQ(sim.get_model().step_size, 1.0); // set by simulation
+
+    EXPECT_EQ(sim.get_result().get_num_time_points(), 2); // stores initial value and single step
+
+    auto expected_result = Eigen::Vector3d{0, 1, 2}; // flows each get capped to y[i]/step_size == 1/1
+    EXPECT_EQ(sim.get_result().get_last_value(), expected_result);
+}
+
+TEST(TestSdeSir, FlowSimulation)
+{
+    // make a single integration step via a flow simulation
+    // this should overwrite the model step size
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(2))
+        .WillOnce(testing::Return(.5))
+        .WillOnce(testing::Return(.5));
+
+    auto integrator = std::make_shared<mio::EulerIntegratorCore>();
+    auto sim        = mio::ssir::FlowSimulation(ssir_testing_model(), 0, 1);
+    sim.set_integrator(integrator);
+    sim.advance(1);
+
+    EXPECT_EQ(sim.get_model().step_size, 1.0); // set by simulation
```

Comment:
Don't understand why this is 1.0. Isn't the default 0.1? In the `model` and `Simulation`?